### PR TITLE
Add more exception free wrappers to OrbitBase and some cleanups

### PR DESCRIPTION
**Pull request by dimitry-**

### CODE REVIEW

File: src/OrbitBase/FileTest.cpp [hunk 1]
```diff
@@ -305,6 +305,22 @@ TEST(File, MoveFile) {
   EXPECT_FALSE(exists_or_error.value());
 }
 
+TEST(File, RemoveFile) {
+  auto tmp_file_or_error = TemporaryFile::Create();
+  ASSERT_THAT(tmp_file_or_error, HasNoError());
+  auto tmp_file = std::move(tmp_file_or_error.value());
+
+  // Windows is weird, for some reason it does not allow to move opened files.
```
Comment (beckerhe) [inline src/OrbitBase/FileTest.cpp:313]: This comment is from the `MoveFile` test, isn't it? It makes no sense here

File: src/OrbitBase/File.cpp [hunk 2]
```diff
@@ -231,6 +231,16 @@ ErrorMessageOr<void> RemoveFile(const std::filesystem::path& file_path) {
   return outcome::success();
 }
 
+ErrorMessageOr<void> CreateDirectory(const std::filesystem::path& file_path) {
+  std::error_code error;
+  std::filesystem::create_directories(file_path, error);
```
Comment (beckerhe) [inline src/OrbitBase/File.cpp:236]: Have you considered `ErrorMessageOr<bool>` as the return type? `create_directories` will return false if the directory already exists. This might be useful in some situations.

**Review by beckerhe — APPROVED** [review]
Otherwise LGTM

File: src/Orbit/main.cpp [hunk 3]
```diff
@@ -157,7 +159,7 @@ static QStringList ExtractCommandLineFlags(const std::vector<std::string>& comma
   QStringList command_line_flags;
   absl::flat_hash_set<std::string> positional_arg_set(positional_args.begin(),
                                                       positional_args.end());
-  for (std::string command_line_arg : command_line_args) {
+  for (const std::string& command_line_arg : command_line_args) {
```
Comment (vickyliu-go4it) [inline src/Orbit/main.cpp:162]: Nit: Maybe use `string_view` instread of `const std::string&`.

**Review by vickyliu-go4it — APPROVED** [review]
LGTM! Only some minor suggestions : )

File: src/Orbit/main.cpp (hunk 3 above)
Comment (dimitry-) [reply src/Orbit/main.cpp:162]: command_line_args is a vector of strings.. what are advantages of using string_view in this case?

File: src/OrbitBase/File.cpp (hunk 2 above)
Comment (dimitry-) [reply src/OrbitBase/File.cpp:236]: done.

File: src/OrbitBase/FileTest.cpp (hunk 1 above)
Comment (dimitry-) [reply src/OrbitBase/FileTest.cpp:313]: fixed comment.

File: src/Orbit/main.cpp (hunk 3 above)
Comment (vickyliu-go4it) [reply src/Orbit/main.cpp:162]: I was thinking that `const std::string&` can be replace by `string_view` as the latter one would be faster in general cases (go/totw/179). 

But I agree that there is no advantage in this case.

File: src/OrbitBase/File.cpp [hunk 4]
```diff
@@ -221,6 +221,26 @@ ErrorMessageOr<void> MoveFile(const std::filesystem::path& from, const std::file
   return outcome::success();
 }
 
+ErrorMessageOr<void> RemoveFile(const std::filesystem::path& file_path) {
```
Comment (dpallotti) [inline src/OrbitBase/File.cpp:224]: Building on Henning's other comment, if I understand correctly `remove` also returns a `bool`. If I understand correctly, when the file didn't exists then it returns false without setting the error.

File: src/OrbitCore/Path.cpp [hunk 5]
```diff
@@ -33,9 +34,17 @@ static std::string GetEnvVar(const char* variable_name) {
   return var;
 }
 
+static void CreateDirectoryOrDie(const std::filesystem::path& directory) {
+  auto create_directory_result = orbit_base::CreateDirectory(directory);
+  if (create_directory_result.has_error()) {
+    FATAL("Unable to create directory \"%s\": %s", directory.string(),
```
Comment (dpallotti) [inline src/OrbitCore/Path.cpp:40]: Are we fine with just crashing in all the functions below?

File: src/OrbitBase/FileTest.cpp [hunk 6]
```diff
@@ -305,6 +305,40 @@ TEST(File, MoveFile) {
   EXPECT_FALSE(exists_or_error.value());
 }
 
+TEST(File, RemoveFile) {
+  auto tmp_file_or_error = TemporaryFile::Create();
+  ASSERT_THAT(tmp_file_or_error, HasNoError());
+  auto tmp_file = std::move(tmp_file_or_error.value());
+
+  // Windows is weird, for some reason it does not allow to remove opened files.
```
Comment (dpallotti) [inline src/OrbitBase/FileTest.cpp:313]: That's weird from a Linux perspective, but I've seen the "file is open in another program" error often enough to not find this weird at all. 😀 (Nothing to do here anyway.)

**Review by dpallotti — APPROVED** [review]

File: src/OrbitCore/Path.cpp (hunk 5 above)
Comment (dimitry-) [reply src/OrbitCore/Path.cpp:40]: we have done it before just implicitly, but I agree there is a better way to do it, we will need to separate create and get have create return ErrorMessageOr and call initialization from somewhere we can display the error. Orbit should still probably not start but at least user will know why and it won't be a crash.

But this is for another PR, I did not want to do it now.

File: src/OrbitBase/File.cpp (hunk 4 above)
Comment (dimitry-) [reply src/OrbitBase/File.cpp:224]: right.

Comment (dimitry-) [reply src/OrbitBase/File.cpp]: done.

File: src/OrbitQt/MoveFilesDialog.ui [hunk 7]
```diff
@@ -0,0 +1,84 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (dpallotti) [inline src/OrbitQt/MoveFilesDialog.ui:1]: This file doesn't belong in this change, does it?

File: src/OrbitBase/FileTest.cpp (hunk 6 above)
Comment (dpallotti) [reply src/OrbitBase/FileTest.cpp]: In a way this comment could have stayed, it was still helpful. Maybe just "Windows doesn't allow to remove opened files."

File: src/OrbitQt/MoveFilesDialog.ui (hunk 7 above)
Comment (dimitry-) [reply src/OrbitQt/MoveFilesDialog.ui:1]: ouch... right sorry. removing.

File: src/OrbitBase/FileTest.cpp (hunk 6 above)
Comment (dimitry-) [reply src/OrbitBase/FileTest.cpp]: removing the file it is now part of the test-case we want to make sure RemoveFile on non-existing file returns false.

Comment (dpallotti) [reply src/OrbitBase/FileTest.cpp]: Ah yeah, sorry.
